patches/tModLoader/Terraria/ModLoader/Default/UnloadedTilesSystem.cs

```diff
@@ -5,113 +5,49 @@
 namespace Terraria.ModLoader.Default
 {
 	[LegacyName("UnloadedTilesWorld")]
-	class UnloadedTilesSystem : ModSystem
-	{
-		internal List<UnloadedTileInfo> infos = new List<UnloadedTileInfo>();
-		internal List<UnloadedTileInfo> pendingInfos = new List<UnloadedTileInfo>();
+	internal class UnloadedTilesSystem : ModSystem {
+		/// <summary>
+		/// Tile-<see cref="UnloadedInfo"/>s that are not able to be restored in the current state of the world (and saved for the next world load)
+		/// </summary>
+		internal List<UnloadedInfo> tileInfos = new List<UnloadedInfo>();
+
+		/// <summary>
+		/// Wall-<see cref="UnloadedInfo"/>s that are not able to be restored in the current state of the world (and saved for the next world load)
+		/// </summary>
+		internal List<UnloadedInfo> wallInfos = new List<UnloadedInfo>();
 
 		public override void OnWorldLoad() {
-			infos.Clear();
-			pendingInfos.Clear();
+			tileInfos.Clear();
+			wallInfos.Clear();
 		}
 
 		public override TagCompound SaveWorldData() {
 			return new TagCompound {
-				["list"] = infos.Select(info => info?.Save() ?? new TagCompound()).ToList()
+				["tileInfos"] = tileInfos.Select(info => info?.Save() ?? new TagCompound()).ToList(),
```

Comment:
null infos are allowed?

Reply:
Yes, the length of the list of infos never changes, "recovered" data is just nulled out. this keeps indexes consistent

Reply:
Thanks for responding Dire - Expounding a bit, the only usage is having fixed indices being stored in the mapping. 
The nulled entries "should" then be reusable as posIndexing should use null entries instead of adding net new.

Reply:
The entries shouldn't be null anymore, as tracking indexes are all run time now.
I've kept the null check for now, but it *should* be safe to remove, assuming everything works.